jupyter_matlab_proxy/app.py

```diff
@@ -398,4 +399,18 @@ def main():
         runner, host=app["settings"]["host_interface"], port=app["settings"]["app_port"]
     )
     loop.run_until_complete(site.start())
-    loop.run_forever()
+    
+    fu = loop.create_future()
+    # on SIGTERM setting the future result allowing the loop to exit.
+    loop.add_signal_handler(signal.SIGTERM, lambda : fu.set_result(True))
+    loop.run_until_complete(fu)
+
+    async def shutdown():
+        logger.info("Shutting down MATLAB proxy-app")
+        await app.shutdown()
+        await app.cleanup()
+        # waiting here to allow matlab to finish exiting.
+        await asyncio.sleep(5)
```

Comment:
Is this required? looks like the app cleanup is waiting for the MATLAB process to exist

Reply:
I think so. i tried several times and in few cases matlab did not completed the exit and I got locked out of the matworks account for about half an hour. I will verify again.

Reply:
verified. without this delay matlab not finishing properly and the user is locked out.